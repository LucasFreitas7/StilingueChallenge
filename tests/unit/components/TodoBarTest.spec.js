import { shallowMount } from '@vue/test-utils'


const toggleThemeMock = jest.fn()
describe('TodoBar.vue', () => {
  const wrapper = shallowMount(TodoBar)
  it('Quando clicado no botão deve trocar o tema da aplicação', () => {
    wrapper.setMethods({toggleTheme: toggleThemeMock})
    wrapper.find('img').trigger('click')
    console.log(wrapper.html())
    expect(toggleThemeMock).toHaveBeenCalled()
  })
  it('teste de snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

})
