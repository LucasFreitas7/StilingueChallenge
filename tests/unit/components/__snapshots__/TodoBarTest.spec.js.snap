// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoBar.vue teste de snapshot 1`] = `
<div class="top-bar">
  <h1>TODO</h1> <img src="" alt="button-light-theme">
</div>
`;
